core/controllers/cron.py

```diff
@@ -230,6 +230,23 @@ def get(self):
             logging.warning('Deletion jobs for JobModels entities kicked off.')
 
 
+class CronModelsCleanupHandler(base.BaseHandler):
+    """Handler for cleaning up models that are marked as deleted and are and
+    marking specific types of models as deleted.
+    """
+
+    @acl_decorators.can_perform_cron_tasks
+    def get(self):
+        """Cron handler that is run every week and hard-deletes all models that
+        are marked as deleted (have deleted field set to True) and were
+        last updated more than some period of time. Also, for some types of
+        models (that we shouldn't keep for long time) mark them as deleted if
+        they were last updated more than some period of time ago.
+        """
+        cron_services.delete_models_marked_as_deleted()
+        cron_services.mark_models_as_deleted()
```

Comment:
Should we run this in taskqueue? Considering it can take a lot of time and there can be a timeout issue for this handler? (not sure!)

Reply:
This will work like general API, right? So we have a time limit on completing one request. is there any limitation on the number of models we have to delete through each call? [Like do we expect that there won't be more than X UserQuery model to delete after 4 weeks?]

Reply:
Nvm, I found that now we are using delete_multi so there won't be an issue (somewhat!)